lib/metasploit/framework/data_service/remote/http/remote_host_data_service.rb

```diff
@@ -23,6 +23,15 @@ def report_hosts(hosts)
     self.post_data(HOST_API_PATH, hosts)
   end
 
+  def update_host(opts)
+    path = HOST_API_PATH
+    if opts && opts[:id]
```

Comment:
since you're checking for existence should you throw e if these things are missing

Reply:
The check is used to build the `id` into the URL path. It is not an error here as future bulk updates would be performed against the `/api/v1/hosts` endpoint rather than `/api/v1/hosts/identifier`. An error should still occur if an `id` is unavailable to the servlet.